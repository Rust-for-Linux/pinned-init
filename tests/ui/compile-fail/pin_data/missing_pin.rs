extern crate pin_init;
use pin_init::*;

#[pin_data]
struct Foo {
    a: usize,
}

impl Foo {
    fn new(a: impl PinInit<usize>) -> impl PinInit<Self> {
        pin_init!(Self {
            a <- a,
        })
    }
}

fn main() {}
