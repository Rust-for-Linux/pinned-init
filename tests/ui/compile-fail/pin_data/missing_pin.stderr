error[E0277]: the trait bound `impl PinInit<usize>: Init<usize, _>` is not satisfied
  --> tests/ui/compile-fail/pin_data/missing_pin.rs:11:9
   |
11 | /         pin_init!(Self {
12 | |             a <- a,
   | |             - required by a bound introduced by this call
13 | |         })
   | |__________^ the trait `Init<usize, _>` is not implemented for `impl PinInit<usize>`
   |
   = help: the trait `Init<usize, _>` is not implemented for `impl PinInit<usize>`
           but trait `Init<impl PinInit<usize>, _>` is implemented for it
   = help: for that trait implementation, expected `impl PinInit<usize>`, found `usize`
note: required by a bound in `__ThePinData::a`
  --> tests/ui/compile-fail/pin_data/missing_pin.rs:4:1
   |
4  | #[pin_data]
   | ^^^^^^^^^^^ required by this bound in `__ThePinData::a`
5  | struct Foo {
6  |     a: usize,
   |     - required by a bound in this associated function
   = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
